crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4108,16 +4124,10 @@ impl<'db> Class<'db> {
                     let bindings = use_def.public_bindings(symbol);
                     let inferred = symbol_from_bindings(db, bindings);
 
-                    match inferred {
-                        Symbol::Type(ty, _) => SymbolAndQualifiers(
-                            Symbol::Type(
-                                UnionType::from_elements(db, [Type::unknown(), ty]),
-                                Boundness::Bound,
-                            ),
-                            qualifiers,
-                        ),
-                        Symbol::Unbound => SymbolAndQualifiers(Symbol::Unbound, qualifiers),
-                    }
+                    SymbolAndQualifiers(
+                        inferred.map_type(|ty| UnionType::from_elements(db, [Type::unknown(), ty])),
+                        qualifiers,
+                    )
```

Comment:
TODO for me: special casing for `Final` symbols (etc) should also be done here (just like in `symbol`)